homeassistant/components/sia/alarm_control_panel.py

```diff
@@ -0,0 +1,285 @@
+    STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import (
+    CONF_ACCOUNT,
+    CONF_ACCOUNTS,
+    CONF_PING_INTERVAL,
+    CONF_ZONES,
+    DATA_UPDATED,
+    DOMAIN,
+    EVENT_ACCOUNT,
+    EVENT_CODE,
+    EVENT_ID,
+    EVENT_MESSAGE,
+    EVENT_TIMESTAMP,
+    EVENT_ZONE,
+    PING_INTERVAL_MARGIN,
+    SIA_EVENT,
+)
+from .utils import get_attr_from_sia_event, get_entity_and_name, get_ping_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+DEVICE_CLASS_ALARM = "alarm"
+PREVIOUS_STATE = "previous_state"
+
+CODE_CONSEQUENCES = {
+    "PA": STATE_ALARM_TRIGGERED,
+    "JA": STATE_ALARM_TRIGGERED,
+    "TA": STATE_ALARM_TRIGGERED,
+    "BA": STATE_ALARM_TRIGGERED,
+    "CA": STATE_ALARM_ARMED_AWAY,
+    "CG": STATE_ALARM_ARMED_AWAY,
+    "CL": STATE_ALARM_ARMED_AWAY,
+    "CP": STATE_ALARM_ARMED_AWAY,
+    "CQ": STATE_ALARM_ARMED_AWAY,
+    "CS": STATE_ALARM_ARMED_AWAY,
+    "CF": STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    "OA": STATE_ALARM_DISARMED,
+    "OG": STATE_ALARM_DISARMED,
+    "OP": STATE_ALARM_DISARMED,
+    "OQ": STATE_ALARM_DISARMED,
+    "OR": STATE_ALARM_DISARMED,
+    "OS": STATE_ALARM_DISARMED,
+    "NC": STATE_ALARM_ARMED_NIGHT,
+    "NL": STATE_ALARM_ARMED_NIGHT,
+    "BR": PREVIOUS_STATE,
+    "NP": PREVIOUS_STATE,
+    "NO": PREVIOUS_STATE,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: Callable[[], None]
+) -> bool:
+    """Set up SIA alarm_control_panel(s) from a config entry."""
+    async_add_entities(
+        [
+            SIAAlarmControlPanel(
+                *get_entity_and_name(
+                    entry.data[CONF_PORT], acc[CONF_ACCOUNT], zone, DEVICE_CLASS_ALARM
+                ),
+                entry.data[CONF_PORT],
+                acc[CONF_ACCOUNT],
+                zone,
+                acc[CONF_PING_INTERVAL],
+            )
+            for acc in entry.data[CONF_ACCOUNTS]
+            for zone in range(1, acc[CONF_ZONES] + 1)
+        ]
+    )
+    return True
+
+
+class SIAAlarmControlPanel(AlarmControlPanelEntity, RestoreEntity):
+    """Class for SIA Alarm Control Panels."""
+
+    def __init__(
+        self,
+        entity_id: str,
+        name: str,
+        port: int,
+        account: str,
+        zone: int,
+        ping_interval: int,
+    ):
+        """Create SIAAlarmControlPanel object."""
+        self.entity_id = ALARM_FORMAT.format(entity_id)
+        self._unique_id = entity_id
+        self._name = name
+        self._port = port
+        self._account = account
+        self._zone = zone
+        self._ping_interval = get_ping_interval(ping_interval)
```

Comment:
Please remove side effects from init

Reply:
I changed it, but I'm not sure how you define a side-effect, the get_ping_interval function is basically a cast of the int to a timedelta, so is casting also considered a side effect?